etcdserver/auth/auth.go

```diff
@@ -281,11 +285,14 @@ func (s *store) UpdateUser(user User) (User, error) {
 		return old, err
 	}
 
-	hash, err := s.HashPassword(user.Password)
-	if err != nil {
-		return old, err
+	if user.Password != "" {
+		var hash string
+		hash, err = s.HashPassword(user.Password)
```

Comment:
So hash(empty) is not empty? And with your change, do we need that in merge at all? Or should we move your change to merge? I do not want the check to be at multiple places and one of them is wrong.